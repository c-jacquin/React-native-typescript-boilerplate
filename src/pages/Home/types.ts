export interface HomeProps {}

export interface HomeState {
    items: any[]
}

export interface HomeActionCreators {}

export interface HomeConnectedProps {}
