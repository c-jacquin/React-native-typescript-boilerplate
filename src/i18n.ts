import 'intl'
import { addLocaleData } from 'react-intl'

import enLocaleData from 'react-intl/locale-data/en'
import frLocaleData from 'react-intl/locale-data/fr'

import enTranslationMessages from '../i18n/en.json'
import frTranslationMessages from '../i18n/fr.json'

import config from 'config'

/**
 * add helpers to format date strings and all the i18n magic
 */
addLocaleData(enLocaleData)
addLocaleData(frLocaleData)

export interface TranslationMessages {
    en: any
    fr: any
}

export const formatTranslationMessages = (
    locale: string,
    messages: any
): TranslationMessages => {
    const defaultFormattedMessages: any =
        locale !== config.LANGUAGE.DEFAULT_LOCALE
            ? formatTranslationMessages(
                  config.LANGUAGE.DEFAULT_LOCALE,
                  enTranslationMessages
              )
            : {}
    const formattedMessages: TranslationMessages | any = {}
    const messageKeys = Object.keys(messages)

    for (const messageKey of messageKeys) {
        formattedMessages[messageKey] =
            locale === config.LANGUAGE.DEFAULT_LOCALE
                ? messages[messageKey]
                : (formattedMessages[messageKey] =
                      messages[messageKey] ||
                      defaultFormattedMessages[messageKey])
    }

    return formattedMessages
}

export const translationMessages: TranslationMessages = {
    en: formatTranslationMessages('en', enTranslationMessages),
    fr: formatTranslationMessages('fr', frTranslationMessages),
}
