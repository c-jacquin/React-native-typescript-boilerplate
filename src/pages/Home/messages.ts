import { defineMessages } from 'react-intl'

export default defineMessages({
    title: {
        id: 'app.title',
    },
})
