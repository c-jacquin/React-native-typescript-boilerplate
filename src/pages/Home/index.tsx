import React, { Component } from 'react'
import { ScrollView, Text } from 'react-native'
import { connect, MapStateToProps, MapDispatchToProps } from 'react-redux'
import { bindActionCreators, Dispatch } from 'redux'

import { FormattedMessage } from 'components/intl'
import { AppState, ReduxAction } from 'store/types'

import {
    HomeProps,
    HomeState,
    HomeConnectedProps,
    HomeActionCreators,
} from './types'
import messages from './messages'

export class Home extends Component<HomeProps, HomeState> {
    static navigationOptions = {
        title: 'Home',
    }

    render() {
        return (
            <ScrollView>
                <FormattedMessage {...messages.title} />
            </ScrollView>
        )
    }
}

const mapStateToProps: MapStateToProps<
    HomeConnectedProps,
    HomeProps,
    AppState
> = state => ({})

const mapDispatchToProps: MapDispatchToProps<HomeActionCreators, HomeProps> = (
    dispatch: Dispatch<ReduxAction>
) => bindActionCreators({}, dispatch)

export default connect<
    HomeConnectedProps,
    HomeActionCreators,
    HomeProps,
    AppState
>(mapStateToProps, mapDispatchToProps)(Home)
