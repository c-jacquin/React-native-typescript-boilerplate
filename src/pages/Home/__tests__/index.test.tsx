import 'rxjs'
import * as React from 'react'
import 'react-native'
import { Provider } from 'react-redux'
import { IntlProvider } from 'react-intl'
import * as renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import Home from '../index'

import { translationMessages } from '../../../i18n'

const mockStore = configureStore()

describe('Home component', () => {
    let store = mockStore({})
    beforeEach(() => {
        store = mockStore({})
    })

    it('should renders correctly', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <IntlProvider
                        locale={'en'}
                        messages={translationMessages.en}
                    >
                        <Home />
                    </IntlProvider>
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
