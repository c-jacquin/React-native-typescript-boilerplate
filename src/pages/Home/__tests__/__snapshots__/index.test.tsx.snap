// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Hello World
    </Text>
  </View>
</RCTScrollView>
`;
